transmogrifier/sources/xml/ead.py

```diff
@@ -362,19 +327,86 @@ def generate_name_identifier_url(cls, name_element: Tag) -> list | None:
         return None
 
     @classmethod
-    def get_main_titles(cls, xml: Tag) -> list[str]:
+    def _get_collection_description(cls, source_record: Tag) -> Tag:
+        if collection_description := source_record.metadata.find(
+            "archdesc", level="collection"
+        ):
+            return collection_description
+        message = (
+            "Record skipped because key information is missing: "
+            '<archdesc level="collection">.'
+        )
+        raise SkippedRecordEvent(message)
+
+    @classmethod
+    def _get_collection_description_did(cls, source_record: Tag) -> Tag:
+        collection_description = cls._get_collection_description(source_record)
+        if collection_description_did := collection_description.did:
+            return collection_description_did
+        message = "Record skipped because key information is missing: <did>."
+        raise SkippedRecordEvent(message)
+
+    @classmethod
+    def _get_control_access(cls, source_record: Tag) -> list[Tag]:
+        collection_description = cls._get_collection_description(source_record)
+        return collection_description.find_all("controlaccess", recursive=False)
+
+    @classmethod
+    def get_alternate_titles(
+        cls, source_record: Tag
+    ) -> list[timdex.AlternateTitle] | None:
+        """Retrieve extra titles from get_main_titles."""
+        return [
+            timdex.AlternateTitle(value=title)
+            for index, title in enumerate(cls.get_main_titles(source_record))
+            if index > 0 and title
+        ] or None
+
+    @classmethod
+    def get_citation(cls, source_record: Tag) -> str | None:
+        collection_description = cls._get_collection_description(source_record)
+        if (
+            citation_element := collection_description.find("prefercite", recursive=False)
+        ) and (
+            citation := cls.create_string_from_mixed_value(
+                citation_element, " ", ["head"]
```

Comment:
We may want to start using named args for this

Reply:
I was considering this but felt conflicted [with @ghukill 's comment on a previous PR](https://github.com/MITLibraries/transmogrifier/pull/177#discussion_r1613894603), but perhaps the guideline @ghukill is only applicable for "methods with single positional arguments" or "when testing"?

In any case, I will update to use named args! I think `create_string_from_mixed_value` is nifty but is difficult to understand just by the name alone.

Reply:
To clarify my comment a bit...

If I'm reading correctly, this is the signature:
```python
def create_string_from_mixed_value(
        cls,
        xml_element: Tag,
        separator: str = "",
        skipped_elements: list[str] | None = None,
    )...
```

So anytime called, I'd expect to see:
  - required, positional argument `xml_element` is not named, but just passed positionally
  - the rest of them, passed as name arguments, because named in the signature
    - _unless_ the defaults are okay, and then okay to omit

Does that mesh with the discussion here?

Reply:
So in this case, I'd expect:

```python
cls.create_string_from_mixed_value(
    citation_element,
    separator = " ",
    skipped_elements = ["head"]
)
```

Open to pushback, but this convention feels like a pretty widely followed convention?

Reply:
Hmm, I think my confusion was I took "named arguments" (or keyword arguments) literally. 🤔 Technically, all the arguments are positional since the signature isn't as follows:

```python
def create_string_from_mixed_value(
        cls,
        xml_element: Tag,
        *,
        separator: str = "",
        skipped_elements: list[str] | None = None,
)
```

I'm definitely on board with the proposed convention; just curious if it's more accurate to say: 

> if overriding default positional arguments, use keyword arguments in the method call
?

Reply:
If I'm reading it right, this is the actual function definition:

```python
@classmethod
def create_string_from_mixed_value(
    cls,
    xml_element: Tag,
    separator: str = "",
    skipped_elements: list[str] | None = None,
) -> str:
   ...
```

Apologies if I'm being imprecise with my words here or earlier, but I would characterize the arguments like:
  * `cls`: required, placeholder, can kind of ignore
  * `xml_element`: required, positional
  * `separator`: because it has equals sign `=`, it's named, optional, with default
  * `skipped_elements`: same as separator

I take your point that _technically_ all the arguments are positional, good point.  But I suppose my position is that _conventionally_ -- and this is open to debate for sure -- if an argument has an equals sign `=` than it's treated as a named argument, and therefore optional with a default value.

This [SO post](https://stackoverflow.com/a/57819001) breaks it down nicely, and exposes that I _have_ been lazy with my terminology.

If we ever do deal with functions that use `*args` or `**kwargs`, then I think it gets more subtle (to your example above), but for most signatures, I feel like it's often positional (no defaults) and then named/keyword (with defaults).  My vote is to
   * a) _not_ use the arg name for positionals
   * b) use the arg name for named/keyword

Thoughts?  or am I just making a mountain out of a mole hill 😅?

Reply:
Thank you for clarifying! I accept / understand your definitions and we are on the same page! I do wonder whether we should start using `*` or `/` in our function signatures to more explicitly distinguish positional vs named/keyword arguments but beyond the scope of this PR. 🤓 If I remember correctly, [Ruff throws errors for boolean positional arguments (FBT001)](https://docs.astral.sh/ruff/rules/boolean-type-hint-positional-argument/).

Circling back to original comment: [changes have been made](https://github.com/MITLibraries/transmogrifier/pull/185/commits/de46532bbe2390a48aed2927793e60a130ba0a50).

Reply:
That feels right to me but I honestly am learning some of these conventions in these PR conversations 🙃

Reply:
This conversation is the first time I've seen the `/` operator for function/method signatures, and helped clarify for me what the `*` operator does _without_ an immediate following variable name like `*args`.  

This helped me understand it more tacitly:

```python
In [1]: def foo(a, b, *, c="horse"):
   ...:     print(a,b,c)
   ...: 

In [2]: foo(42, 101, 999, 888, c='zebra')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[2], line 1
----> 1 foo(42, 101, 999, 888, c='zebra')

TypeError: foo() takes 2 positional arguments but 4 positional arguments (and 1 keyword-only argument) were given

In [3]: def bar(a, b, *args, c="horse"):
   ...:     print(a,b,args,c)
   ...: 

In [4]: bar(42, 101, 999, 888, c='zebra')
42 101 (999, 888) zebra
```

I'm definitely not opposed to using `*` and `/` in signatures if we find them useful.  Here is an edge case where I could see it being useful, when you want to force keyword argument or no positional:

```python
def baz(*, a=1):
    pass

In [6]: baz()
1

In [7]: baz(42)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[7], line 1
----> 1 baz(42)

TypeError: baz() takes 0 positional arguments but 1 was given

In [8]: baz(a=101)
101    
```

Ineresting stuff!  In cases without `*` or `/`, glad we're honing in on some conventions, and curious if we ever run into situations where it would be nice to use them.  I read an article this morning that I found pretty convincing that the main appeal is often APIs, where you want to finely control how functions are called by users potentially not familiar with the library.